### PR TITLE
[ML] Make ml internal indices hidden

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR makes the following indices hidden:
1. `.ml-anomalies*`
2. `.ml-state*`
3. `.ml-notifications*`
4. `.ml-annotations*`

Transform indices (`.transform-notifications-*` and deprecated `.data-frame-notifications-*`) and inference index (`.ml-inference-000001`) are *not* converted in this PR.

Relates https://github.com/elastic/elasticsearch/issues/52420

### CODE REVIEW

**Comment by droberts195** [comment]
I think line 941 of `MachineLearning.java` also needs deleting.  At the time it was added it must have been expected that `.ml-state` would become a system index, not a hidden index.  But since we want rollover type behaviour on it we've now decided that it will be a hidden index.

**Review by benwtrent — COMMENTED** [review]
Do we need to worry about things like the `ExpiredForecastsRemover`? 
It hits `AnomalyDetectorsIndex.jobResultsIndexPrefix() + "*";` for its search but does not set the expansion flags to include hidden indices.

**Comment by przemekwitek** [comment]
> I think line 941 of `MachineLearning.java` also needs deleting. At the time it was added it must have been expected that `.ml-state` would become a system index, not a hidden index. But since we want rollover type behaviour on it we've now decided that it will be a hidden index.

Done.

File: x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/notifications_index_template.json [hunk 1]
```diff
@@ -7,8 +7,9 @@
   "settings" : {
     "index" : {
       "number_of_shards" : "1",
-      "auto_expand_replicas" : "0-1"
-    }
+      "auto_expand_replicas" : "0-1",
+      "hidden": true
+}
```
[the remark this answers is not in the document]
Comment (przemekwitek) [reply x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/notifications_index_template.json]: Done.

**Review by przemekwitek — COMMENTED** [review]
> Do we need to worry about things like the ExpiredForecastsRemover?
It hits AnomalyDetectorsIndex.jobResultsIndexPrefix() + "*"; for its search but does not set the expansion flags to include hidden indices.

My understanding is that it will work correctly as it provides the pattern starting with a dot ('.'). The problem occurs when (like in `MlDistributedFailureIT.java`) no index is provided.

**Comment by droberts195** [comment]
> It hits `AnomalyDetectorsIndex.jobResultsIndexPrefix() + "*";` for its search but does not set the expansion flags to include hidden indices.

My understanding is that wildcard patterns still match hidden indices providing the pattern begins with a dot - see the second bullet point in the description of #50452.  So we _should_ be OK.

Once this PR is merged the whole team will need to be on the lookout for unforeseen side effects.  It's good that we're not doing this close to feature freeze.

**Review by droberts195 — APPROVED** [review]
LGTM

**Review by benwtrent — APPROVED** [review]
👍 cool, glad hidden are still expanded if their prefix if specified.

**Comment by przemekwitek** [comment]
run elasticsearch-ci/2

**Comment by przemekwitek** [comment]
run elasticsearch-ci/2

**Pull request merged**